Do not try to lint submodules

kalkin opened the pull request:
Submodules are marked as changed by `git diff-index` and AFAIK there is
not way to filter them out via `--diff-filter` argument

sk- commented:
Why is this change required? I think in case of directories, the output should say that no linter was found for such a file. Or is it crashing? If so, please file a bug report.

I would like to know whether there's a valid case in which a folder could be returned by git or mercurial for example. Otherwise, we probably would need to add a method to the vcs objects, asking whether the file is a submodule (part of the repo or not).

Also, this PR is missing tests.

kalkin commented:
Forget it. It's a bug in my `git-hook`, which I introduced, then fixed at the same time as I added this changes.

Basically if you do `git lint ""`  it will fail. Not sure if you should handle this case.
